tests/unit/amo/sagas/test_addonsByAuthors.js

```diff
@@ -0,0 +1,133 @@
+import SagaTester from 'redux-saga-tester';
+
+import addonsByAuthorsReducer, {
+  OTHER_ADDONS_BY_AUTHORS_PAGE_SIZE,
+  fetchOtherAddonsByAuthors,
+  loadOtherAddonsByAuthors,
+} from 'amo/reducers/addonsByAuthors';
+import addonsByAuthorsSaga from 'amo/sagas/addonsByAuthors';
+import {
+  ADDON_TYPE_THEME,
+  CLEAR_ERROR,
+  SEARCH_SORT_POPULAR,
+} from 'core/constants';
+import * as searchApi from 'core/api/search';
+import apiReducer from 'core/reducers/api';
+import {
+  createAddonsApiResult,
+  dispatchClientMetadata,
+  fakeAddon,
+} from 'tests/unit/amo/helpers';
+import { createStubErrorHandler } from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  let errorHandler;
+  let mockApi;
+  let sagaTester;
+
+  beforeEach(() => {
+    errorHandler = createStubErrorHandler();
+    mockApi = sinon.mock(searchApi);
+    sagaTester = new SagaTester({
+      initialState: dispatchClientMetadata().state,
+      reducers: {
+        api: apiReducer,
+        addonsByAuthors: addonsByAuthorsReducer,
+      },
+    });
+    sagaTester.start(addonsByAuthorsSaga);
+  });
+
+  function _fetchOtherAddonsByAuthors(params) {
+    sagaTester.dispatch(fetchOtherAddonsByAuthors({
+      errorHandlerId: errorHandler.id,
+      addonType: ADDON_TYPE_THEME,
+      ...params,
+    }));
+  }
+
+  it('calls the API to retrieve other add-ons', async () => {
+    const addons = [fakeAddon];
+    const authors = ['mozilla', 'johnedoe'];
+    const slug = fakeAddon.slug;
+    const state = sagaTester.getState();
+
+    mockApi
+      .expects('search')
+      .withArgs({
+        api: state.api,
+        filters: {
+          addonType: ADDON_TYPE_THEME,
+          author: authors.join(','),
+          page_size: OTHER_ADDONS_BY_AUTHORS_PAGE_SIZE + 1,
+          sort: SEARCH_SORT_POPULAR,
+        },
+      })
+      .once()
+      .returns(Promise.resolve(createAddonsApiResult(addons)));
+
+    _fetchOtherAddonsByAuthors({ authors, slug });
+
+    const expectedLoadAction = loadOtherAddonsByAuthors({ addons, slug });
+
+    await sagaTester.waitFor(expectedLoadAction.type);
+    mockApi.verify();
+
+    const loadAction = sagaTester.getCalledActions()[2];
+    expect(loadAction).toEqual(expectedLoadAction);
+  });
+
+  it('clears the error handler', async () => {
+    _fetchOtherAddonsByAuthors({ authors: [], slug: fakeAddon.slug });
+
+    await sagaTester.waitFor(CLEAR_ERROR);
```

Comment:
This might be a bit more future proof as `sagaTester.waitFor(errorHandler.createClearingAction().type)` (although you probably want to create a variable so you can use it in the assertion).